File: VerificadorPalindromo.pas
program VerificadorPalindromo;
uses
    SysUtils, Crt;

function esPalindromo(texto: string): boolean;
var
    textoLimpio: string;
    i, longitud: integer;
begin
    textoLimpio := LowerCase(StringReplace(texto, ' ', '', [rfReplaceAll]));
    
    longitud := Length(textoLimpio);
    for i := 1 to longitud div 2 do
    begin
        if textoLimpio[i] <> textoLimpio[longitud - i + 1] then
        begin
            esPalindromo := false;
            Exit;
        end;
    end;
    esPalindromo := true;
end;

var
    entrada: string;
begin
    ClrScr;
    Writeln('=== VERIFICADOR DE PALÍNDROMOS ===');
    Write('Ingrese un número o palabra: ');
    Readln(entrada);
    
    if esPalindromo(entrada) then
        Writeln( entrada, ''' ES un palíndromo')
    else
        Writeln(entrada, ''' NO es un palíndromo');
    
    Readln; // Esperar entrada antes de cerrar
end.

File: VerificadorPalindromo.c
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <stdbool.h>

bool esPalindromo(char texto[]) {
    char textoLimpio[100];
    int i, j = 0;
    int longitud;
    
    for (i = 0; texto[i] != '\0'; i++) {
        if (texto[i] != ' ') {
            textoLimpio[j++] = tolower(texto[i]);
        }
    }
    textoLimpio[j] = '\0';
    
    longitud = strlen(textoLimpio);
    for (i = 0; i < longitud / 2; i++) {
        if (textoLimpio[i] != textoLimpio[longitud - i - 1]) {
            return false;
        }
    }
    return true;
}

int main() {
    char entrada[100];
    
    printf("=== VERIFICADOR DE PALÍNDROMOS ===\n");
    printf("Ingrese un número o palabra: ");
    fgets(entrada, sizeof(entrada), stdin);
    
    entrada[strcspn(entrada, "\n")] = '\0';
    
    if (esPalindromo(entrada)) {
        printf(" ES un palíndromo\n", entrada);
    } else {
        printf(" NO es un palíndromo\n", entrada);
    }
    
    return 0;
}